### PR TITLE
Enable ASan+UBSan+LSan and TSan

### PR DESCRIPTION
**Pull request by KVS85**
This PR is intended to:
- enable ASan+UBSan+LSan and TSan modes
- fix overall timeout issue when reports weren't generated (for Linux and macOS)
- fix ASan status result wheth Jenkins was red only when tests failed but not when ASan provided non-empty output

### CODE REVIEW

File: scripts/buildArangoDB3.fish [hunk 1]
```diff
@@ -56,10 +56,14 @@ end
 if test "$ASAN" = "On"
   # Suppress leaks detection only during building
   set -gx ASAN_OPTIONS "detect_leaks=0"
+  set -l SANITIZERS "-fsanitize=address -fsanitize=undefined -fsanitize=float-divide-by-zero -fsanitize=leak"
+  if test "$ASAN_MODE" = "TSan"
```
[the remark this answers is not in the document]
Comment (KVS85) [reply scripts/buildArangoDB3.fish]: Fixed.

**Review by naushniki — APPROVED** [review]

**Comment by KVS85** [comment]
Sanitizers (x86-64):
- [ ] devel: https://jenkins.arangodb.biz/job/arangodb-ANY-linux-asan/894/
- [ ] 3.9: https://jenkins.arangodb.biz/job/arangodb-ANY-linux-asan/895/
- [ ] 3.8: https://jenkins.arangodb.biz/job/arangodb-ANY-linux-asan/896/
- [ ] 3.7: https://jenkins.arangodb.biz/job/arangodb-ANY-linux-asan/897/

Sanitizers (aarch64):
- [ ] devel: https://jenkins.arangodb.biz/job/arangodb-ANY-linux-asan.aarch64/2/

PRs:
- [ ] devel: https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/18929/
- [ ] 3.9: https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/18930/
- [ ] 3.8: https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/18931/
- [ ] 3.7: https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/18932/

**Comment by KVS85** [comment]
Sanitizers (x86-64):
- [ ] devel: https://jenkins.arangodb.biz/job/arangodb-ANY-linux-asan/912/
- [ ] 3.9: https://jenkins.arangodb.biz/job/arangodb-ANY-linux-asan/913/
- [ ] 3.8: https://jenkins.arangodb.biz/job/arangodb-ANY-linux-asan/914/
- [ ] 3.7: https://jenkins.arangodb.biz/job/arangodb-ANY-linux-asan/915/

Sanitizers (aarch64):
- [ ] devel: https://jenkins.arangodb.biz/job/arangodb-ANY-linux-asan.aarch64/3/

PRs:
- [ ] devel: https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/19020/
- [ ] 3.9: https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/19021/
- [ ] 3.8: https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/19022/
- [ ] 3.7: https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/19023/

**Comment by KVS85** [comment]
Sanitizers (x86-64):
- [ ] devel: https://jenkins.arangodb.biz/job/arangodb-ANY-linux-san/921/
- [ ] 3.9: https://jenkins.arangodb.biz/job/arangodb-ANY-linux-san/922/
- [ ] 3.8: https://jenkins.arangodb.biz/job/arangodb-ANY-linux-san/923/
- [ ] 3.7: https://jenkins.arangodb.biz/job/arangodb-ANY-linux-san/924/

Sanitizers (aarch64):
- [ ] N/A

PRs:
- [ ] devel: https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/19025/
- [ ] 3.9: https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/19026/
- [ ] 3.8: https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/19027/
- [ ] 3.7: https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/19028/

**Review by fceller — APPROVED** [review]
LGTM

**Pull request merged**